applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h

```diff
@@ -0,0 +1,1015 @@
+            // Getting the integration points
+            ThisIntegrationMethod = itElem->GetIntegrationMethod();
+            const Element::GeometryType::IntegrationPointsArrayType& IntegrationPoints = rThisGeometry.IntegrationPoints(ThisIntegrationMethod);
+            const unsigned int IntegrationPointsNumber = IntegrationPoints.size();
+            
+            // Computing the Jacobian
+            Vector VectorDetJ(IntegrationPointsNumber);
+            rThisGeometry.DeterminantOfJacobian(VectorDetJ,ThisIntegrationMethod);
+            
+            // Getting the CL
+            std::vector<ConstitutiveLaw::Pointer> ConstitutiveLawVector(IntegrationPointsNumber);
+            itElem->GetValueOnIntegrationPoints(CONSTITUTIVE_LAW,ConstitutiveLawVector,CurrentProcessInfo);
+                
+            for (unsigned int iGaussPoint = 0; iGaussPoint < IntegrationPointsNumber; iGaussPoint++ )
+            {
+                const array_1d<double, 3> LocalCoordinates = IntegrationPoints[iGaussPoint].Coordinates();
+                
+                // We compute the corresponding weight
+                const double Weight = VectorDetJ[iGaussPoint] * IntegrationPoints[iGaussPoint].Weight();
+                
+                // We compute the global coordinates
+                array_1d<double, 3> GlobalCoordinates;
+                GlobalCoordinates = rThisGeometry.GlobalCoordinates( GlobalCoordinates, LocalCoordinates );
+                
+                // We create the respective GP
+                PointTypePointer pPoint = PointTypePointer(new PointType(GlobalCoordinates, ConstitutiveLawVector[iGaussPoint], Weight));
+                (ThisPointVector).push_back(pPoint);
+            }
+        }
+        
+        return ThisPointVector;
+    }
+
+    /**
+     * This method interpolate the values of the GP using the CPT method
+     */
+    
+    void InterpolateGaussPointsCPT()
+    {
+        // We Initialize the process info
+        const ProcessInfo& CurrentProcessInfo = mrDestinationMainModelPart.GetProcessInfo();
+     
+        // We initialize the intergration method
+        GeometryData::IntegrationMethod ThisIntegrationMethod;
+        
+        // We update the list of points
+        mPointListOrigin.clear();     
+        
+        mPointListOrigin = CreateGaussPointList(mrOriginMainModelPart);
+
+//         #pragma omp parallel
+//         {
+            // Create a tree
+            // It will use a copy of mNodeList (a std::vector which contains pointers)
+            // Copying the list is required because the tree will reorder it for efficiency
+            KDTree TreePoints(mPointListOrigin.begin(), mPointListOrigin.end(), mBucketSize); 
+            
+            // Iterate over the destination elements
+            ElementsArrayType& pElements = mrDestinationMainModelPart.Elements();
+            auto numElements = pElements.end() - pElements.begin();
+            
+//             #pragma omp for
```

Comment:
same as above

if you want this parallel it should be
#pragma omp **parallel** for (you miss the parallel)